src/ert/storage/mode.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from enum import Enum
+from functools import wraps
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Literal,
+)
+
+if TYPE_CHECKING:
+    from typing_extensions import Concatenate, ParamSpec, TypeVar
+
+ModeLiteral = Literal["", "r", "w"]
+
+
+class ModeError(ValueError):
+    pass
+
+
+class Mode(str, Enum):
+    NONE = ""
+    READ = "r"
```

Comment:
No uses of this. Suggest either implementing it here, or remove this until the PR that does.

Reply:
Removed